lib/knapsack_pro/formatters/time_tracker.rb

```diff
@@ -22,6 +23,13 @@ def initialize(_output)
         @group = {}
         @paths = {}
         @suite_started = now
+        @scheduled_test_file_paths = []
+        @split_by_example_test_file_paths = Set.new
+      end
+
+      def scheduled_test_file_paths=(scheduled_test_file_paths)
+        @scheduled_test_file_paths = scheduled_test_file_paths
+        upsert_split_by_example_test_file_paths
```

Comment:
recommendation

I'd inline the upsert. It's more clear to see it in one place.